MvvmCross-iOSSupport/MvvmCross.iOS.Support.XamarinSidebar/MvxSidebarPresenter.cs

```diff
@@ -33,94 +32,77 @@ protected virtual bool PresentationHintHandler(MvxPanelPresentationHint hint)
             return true;
         }
 
-        public override void Show(MvxViewModelRequest request)
+        protected override void RegisterAttributeTypes()
         {
-            IMvxIosView viewController = Mvx.Resolve<IMvxIosViewCreator>().CreateView(request);
-            Show(viewController);
+            base.RegisterAttributeTypes();
+
+            _attributeTypesToShowMethodDictionary.Add(
+                typeof(MvxSidebarPresentationAttribute),
+                (vc, attribute, request) => ShowSidebarViewController(vc, attribute as MvxSidebarPresentationAttribute, request));
```

Comment:
I think you should make a direct cast here `(MvxSidebarPresentationAttribute)attribute`, or otherwise make a check and return if the attribute is not of type `MvxSidebarPresentationAttribute`.

Now there is a possibility for a `NullReferenceException` which will only occur in the `ShowSidebarViewController` method.